Google.Api.Generator.Rest/PackageEnumStorage.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Generator.Utils;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace Google.Api.Generator.Rest
+{
+    /// <summary>
+    /// The stored values of enums for parameters within a package model.
+    /// These need to be persisted so that we can avoid taking breaking changes if the Discovery doc
+    /// reorders enum values over time.
+    /// </summary>
+    public class PackageEnumStorage
+    {
+        // Note: this is sorted just to keep things idempotent.
+        private readonly SortedDictionary<string, List<string>> _nameToValuesMap;
+
+        private PackageEnumStorage(SortedDictionary<string, List<string>> nameToValuesMap) =>
+            _nameToValuesMap = nameToValuesMap;
+
+        /// <summary>
+        /// Returns the numeric value to use for the given text value. This is drawn from the existing
+        /// values if the text value is already present, or added to the internal representation and
+        /// given the next available numeric value otherwise.
+        /// </summary>
+        /// <param name="key">The enum model key. This must be unique (and always generated in the same way)
+        /// but otherwise has no specific meaning.</param>
+        /// <param name="parameterName">The name of the parameter (as per the Discovery doc)</param>
```

Comment:
nit: This parameter is not present anymore.

Reply:
Fixed, thanks.